### ISSUE TITLE
General fixes

### ISSUE THREAD

**Issue by harshagrawal13** [harshagrawal13/diabeat#1]
- [ ] fix / confirm the implementation of the F1 score, Precision, Recall
- [ ] is having class weights in cross entropy loss helpful or not? (Currently its randomly assigned but need to be figured)
- [ ] learn about model collapse and other ML lingo.
- [ ] Understand torch lightning fully?
   - [ ] does the loss with manual pytorch code equate to loss with lightning code? (Using proper seed)
   - [ ] check how does accumulate_grad_batches, other things with lightning.
- [ ] Overfitting the model. 
   - [ ] does increasing n-peaks per sample make the model better performant.
   - [ ] does increasing the size of model increase the performance? (width, depth of the transformer).
   - [ ] are there other hyper parameters that are important in model overfitting.
   - [ ] Full hyper-parameter search to minimize train loss and does the result surprise you?
- [ ] Once you are happy with the performance on the training data, then implement val losses + metrics and then generalize. 
   - [ ] only then do you do a full hyperparameter search to minimize val loss.
